### PR TITLE
chore: migrate qrcode worker

### PR DESCRIPTION
**Pull request by Jack-Works**

### CODE REVIEW

File: packages/maskbook/src/web-workers/OnDemandWorker.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        console.log(`OnDemandWorker ${this.id}`, ...args)
+    }
+    protected lastUsed = Date.now()
+    protected use(onReady: () => void) {
+        this.lastUsed = Date.now()
+        this.log('kept alive')
+
+        if (this.worker) return onReady()
+        this.worker = new Worker(...this.init)
+        this.worker.addEventListener('error', (e) => this.dispatchEvent(e))
```
Comment (guanbinrui) [inline packages/maskbook/src/web-workers/OnDemandWorker.ts:21]: No need for removing those listeners?

Comment (Jack-Works) [reply packages/maskbook/src/web-workers/OnDemandWorker.ts:21]: After the EventTarget is being recycled, its listeners are also recycled so doesn't count to reference to the Uint8Array.

File: packages/maskbook/src/web-workers/OnDemandWorker.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        this.init = init
+        this.log('created with', ...init)
+    }
+    protected log(...args: any[]) {
+        console.log(`OnDemandWorker ${this.id}`, ...args)
+    }
+    protected lastUsed = Date.now()
+    protected use(onReady: () => void) {
+        this.lastUsed = Date.now()
+        this.log('kept alive')
```
Comment (septs) [inline packages/maskbook/src/web-workers/OnDemandWorker.ts:17]: ???
keep alive?

File: packages/maskbook/src/web-workers/OnDemandWorker.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        this.worker = new Worker(...this.init)
+        this.worker.addEventListener('error', (e) => this.dispatchEvent(e))
+        this.worker.addEventListener('message', (e) => this.dispatchEvent(e))
+        this.worker.addEventListener('messageerror', (e) => this.dispatchEvent(e))
+        setTimeout(onReady, 50)
+    }
+    terminate() {
+        this.worker?.terminate()
+        this.worker = undefined
+        this.log('terminated')
```
Comment (septs) [inline packages/maskbook/src/web-workers/OnDemandWorker.ts:29]: why not with ${this.id}?

**Pull request merged**